core/src/main/java/greencity/filters/RatingStatisticsSpecification.java

```diff
@@ -0,0 +1,143 @@
+package greencity.filters;
+
+import greencity.annotations.RatingCalculationEnum;
+import greencity.entity.RatingStatistics;
+import greencity.entity.User;
+import greencity.service.UserService;
+import java.time.LocalDate;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeParseException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Setter
+@Component
+@Slf4j
+public class RatingStatisticsSpecification implements Specification<RatingStatistics> {
+    private SearchCriteria searchCriteria;
+    private UserService userService;
```

Comment:
`UserService` **will not** be reachable from DAO layer. Please replace by `UserRepo`.